synapse/lib/stormlib/model.py

```diff
@@ -1269,3 +1020,320 @@ async def _riskHasVulnToVulnerable(self, n, nodata=False):
                     await self.copyData(n, proto, overwrite=False)
 
         return retidens
+
+@s_stormtypes.registry.registerLib
+class LibModelMigrations_0_2_31(s_stormtypes.Lib):
+    '''
+    A Storm library with helper functions for the 0.2.31 model it:sec:cpe migration.
+    '''
+    _storm_locals = (
+        {'name': 'listNodes', 'desc': 'Yield queued nodes.',
+         'type': {'type': 'function', '_funcname': '_methListNodes',
+                  'args': (
+                      {'name': 'form', 'type': 'form', 'default': None,
+                       'desc': 'Only show entries matching the specified form.'},
+                      {'name': 'source', 'type': 'str', 'default': None,
+                       'desc': 'Only show entries that were seen by the specified source.'},
+                      {'name': 'offset', 'type': 'int', 'default': 0,
+                       'desc': 'Skip this many entries.'},
+                      {'name': 'size', 'type': 'int', 'default': None,
+                       'desc': 'Only print up to this many entries.'},
+                  ),
+                  'returns': {'name': 'Yields', 'type': 'list',
+                              'desc': 'A tuple of (offset, form, valu, sources) values for the specified node.', }}},
+        {'name': 'printNode', 'desc': 'Print detailed queued node information.',
+         'type': {'type': 'function', '_funcname': '_methPrintNode',
+                  'args': (
+                      {'name': 'offset', 'type': 'into', 'desc': 'The offset of the queued node to print.'},
+                  ),
+                  'returns': {'type': 'null'}}},
+        {'name': 'repairNode', 'desc': 'Repair a queued node.',
+         'type': {'type': 'function', '_funcname': '_methRepairNode',
+                  'args': (
+                      {'name': 'offset', 'type': 'str', 'desc': 'The node queue offset to repair.'},
+                      {'name': 'newvalu', 'type': 'any', 'desc': 'The new (corrected) node value.'},
+                      {'name': 'remove', 'type': 'boolean', 'default': False,
+                       'desc': 'Specify whether to delete the repaired node from the queue.'},
+                  ),
+                  'returns': {'type': 'dict', 'desc': 'The queue node information'}}},
+    )
+    _storm_lib_path = ('model', 'migration', 's', 'model_0_2_31')
+
+    def getObjLocals(self):
+        return {
+            'listNodes': self._methListNodes,
+            'printNode': self._methPrintNode,
+            'repairNode': self._methRepairNode,
+        }
+
+    async def _hasCoreQueue(self, name):
+        queues = await self.runt.snap.core.listCoreQueues()
```

Comment:
I considered both of those options. I try to avoid try/catch if there's a non-exception way of getting the information but that's probably less of a concern in newer python versions. A `Cortex.hasCoreQueue()` should be pretty easy, I'll do that.

Reply:
Hmm. Actually, hasCoreQueue will probably require a stormlib update too. I'll go with the try/catch instead.